### ISSUE TITLE
TX Builder errors

### ISSUE THREAD

**Issue by AlexDochioiu** [spacebudz/lucid#188]
Hey, I seem to encounter some edge case scenario building some transactions.

Code snippet:

```ts
const tx = lucid //
  .newTx()
  .config({ enableChangeSplitting: true })
  .validTo(Date.now() + config.prepareTxTtl * 1000);

_.forEach(opt.destinations, (destination) =>
  tx.payToAddress(destination.address.bech32(), destination.asLucidAssets())
);

const builtTx = await tx.complete();
```

 sending assets
from : `addr1q9fzvmdq6ee7phma59emyuvu00a8hwfvv7lvj9v336atm7ncyp00ukrpfqgv2m0j0gj8rjznk9v5wwjn42wypt3earfsrlzfkv`
to: `addr1qx526449w6a468tltxqeky4em2u24eer8w9mhsfwu4mn0285m2nujlyhr7khzhynnanjfu50lehklzpf3u42qfmjfkxs43ut04`

When I try to send:
```ts
{
    "2d7444cf9e317a12e3eb72bf424fd2a0c8fbafedf10e20bfdb4ad8ab434845444441": 1000000n,
    "0f316948c5975030a964d7fb8dbac4d83438f8e21223538981e42efd4635": 1000000n
}
```
I get: `Missing input or output for some native asset`

When I try to send:
```ts
{
    "2d7444cf9e317a12e3eb72bf424fd2a0c8fbafedf10e20bfdb4ad8ab434845444441": 1000000n,
    "0f316948c5975030a964d7fb8dbac4d83438f8e21223538981e42efd4635": 1000000n,
    "lovelace": 5000000n
}
```
I get: `underflow`

There are enough funds available for the transaction to be successful. Also, the tx building code normally works (is used successfully for most of TXs).

**Comment by AlexDochioiu** [comment]
Those are the 2 existing UTxOs (hex encoded) at the time when error is observed. Dropping them here because the wallet belongs to on of our users, so it's not within my control whether this wallet gets used further. 

```json
[
 "82825820b5aa6522d8140e4a62fc700addaf7d53b53698cc1007967db665b2c261141880028258390152266da0d673e0df7da173b2719c7bfa7bb92c67bec915918ebabdfa78205efe58614810c56df27a2471c853b159473a53aa9c40ae39e8d3821a002d1d74a4581c99d99e341dfd21809058fd1c32d4ce6622df7cde4bfcf292c79bfa63a2504c6f76656c616365436c75623132313701504c6f76656c616365436c75623133303901581caa4a39a4ea445c3b38ad06f54e2c309414e184e7ee0a62203cf347a6a14e436174736f6e436861696e39323401581caa875545cc79cc1e5b593e43783b2acb10a5f0ac9bfae9fc538dd594a14941746d616e3131323701581cadc5716393953403109c335e68c0384238fd19653e960e03afa1fb1fb04f546865526566726573683030333830014f546865526566726573683030353336014f546865526566726573683030363034014f546865526566726573683031313230014f546865526566726573683031323336014f546865526566726573683031393733014f546865526566726573683032303036014f546865526566726573683033353736014f546865526566726573683033363232014f546865526566726573683033383031014f546865526566726573683033383432014f546865526566726573683033383834014f546865526566726573683034353638014f546865526566726573683035303531014f546865526566726573683036383838014f54686552656672657368303732393201",
  "828258206ce247f8a6776ebf69794664138c3516658ca710f72103c8a72257dee4f2e77001825839019fec61fb385fbb1fc25effb272a620483738fa09cd577405854d1eed78205efe58614810c56df27a2471c853b159473a53aa9c40ae39e8d3821a02a05326b819581c0e14267a8020229adc0184dd25fa3174c3f7d6caadcb4425c70e7c04a14a756e736967313437323601581c0f316948c5975030a964d7fb8dbac4d83438f8e21223538981e42efda14246351b000000e8d086f169581c12f9d9446c422caa870948bae1b8844c26c64958943a3954103b034fa14a75676c7962726f33343201581c1cba2915a5c259ee50841f1b297b752df3e172dd9444faf169e3b934a1444f31353401581c2bc31b7881acbd6558b680d42248e201d9c7a2931e278b95e96daab8a3484455534b3031393601484455534b3033313801484455534b3039353501581c2d7444cf9e317a12e3eb72bf424fd2a0c8fbafedf10e20bfdb4ad8aba1464348454444411b000000e379b23ba0581c33b3d36a5bbaf1d803e7c5d359f0c64dde89aac69dfac17a5a7e24aea2575369636b43697479417263616465536e616b653036303301575369636b43697479417263616465536e616b653036343101581c37b84c373c090fc2d4d6e93a1135f18a970bc30ba85aa5e82e7de16ca24847446f7473323435014847446f747332363401581c52cc19c1d6815d3af4063f6630d9769e3fff296a41ef2105d12d5655a14f43726577526f6c654361726432363301581c5ea3a1976f3d0ed63fc36054a2239351c17c61b0bcb63ed17618ddd2a14b747261736869657333303801581c7ea3e7719938f8acd700327d6c9dffd7523a02dce1d9891b5f3a452da151426c6f636b4f776c7353706f747330363501581c840d7e5f8a63337d75cf056c972b62b5764ce75c8bf9e67e9419eff7a1474338426f74323501581c84442385d6c7b1779160db66f8427995843d71948f328352f405bf14a24562643038390145626433363201581cadc5716393953403109c335e68c0384238fd19653e960e03afa1fb1fa24f546865526566726573683037363534014f54686552656672657368303937363301581cb195436b8611daa8e7e70ff516696df21dd3fc1fb501358ff1c50adaa1581c42697446696e7350696e6e61747573426c61636b4f72616e6765363401581cb5907d4e50a0df7f29648c7d5a63757d8a7b765f8ceebd95e584b952a1524c6f74746572795469636b6574323836363801581cc281975562f394761771f13f599881517fa8455946e7e30454de22daa24e474f415454726962653035323733014e474f41545472696265303539323101581cd68739df427e4da4c53e767e85dbda8b0d141e5339e3f69c8829838aa158194346544d414f5662616266697368316f66323255526e6f313601581cd7726f6b882c6fc2ca0cd96c51e5328e1d577d789b085ee0fbe23bf7a1581855676c7942726f546865446566696e69746976653539343101581ce3ae0ca56d51672819db9b78c2636a74ffbc28d2e7e609ea2e7c26aca14f4338436f6c6c65637469626c65323001581ce8209a96a456202276f66224241a703676122d606d208fe464f2e09fa152576861744c69657357697468696e3534333601581ceadb6819293e8a49e50beece93c7198911dfcd20b9613d8b2dfac7efa14a506574626f743031353601581cf06c76a0bd1bcc12813f63793551aa782b0f7aa59bf1c99ddd5eebffa14c536561706c616e653734383901581cf0ff48bbb7bbe9d59a40f1ce90e9e9d0ff5002ec48f232b49ca0fb9aa145667265356801581cf3a65485a9201e979e03aa4fa9f1d544332b7b8440bcd4aa4c6f7d1da153427574746572666c794566666563743131333401",
]
```

**Comment by AlexDochioiu** [comment]
<img width="880" alt="image" src="https://user-images.githubusercontent.com/38853913/235365382-94418696-d000-4e14-88ad-24a00c11a995.png">

The error gets thrown from inside `txBuilder.balance(...)`

**Comment by AlexDochioiu** [comment]
Closing this. The issue is already fixed in Lucid 0.10.2.
